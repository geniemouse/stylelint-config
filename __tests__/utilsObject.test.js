const obj = require("../utils/object");

describe("alphabetizeByKeyName:", () => {
    const { alphabetizeByKeyName } = obj,
        sample = {
            one: 1,
            two: 2,
            three: 3
        },
        expectedOutput = {
            one: 1,
            three: 3,
            two: 2
        };

    test("is a function", () => {
        expect(typeof alphabetizeByKeyName).toBe("function");
    });

    test("will accept an empty object and output the same", () => {
        expect(alphabetizeByKeyName({})).toEqual({});
    });

    test("alphabetizes the key names of an object", () => {
        const result = alphabetizeByKeyName(sample);
        expect(result).toMatchSnapshot();
        expect(Object.keys(result)).toEqual(Object.keys(expectedOutput));
    });
});
