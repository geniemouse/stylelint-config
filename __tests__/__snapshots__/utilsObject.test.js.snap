// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`alphabetizeByKeyName: alphabetizes the key names of an object 1`] = `
Object {
  "one": 1,
  "three": 3,
  "two": 2,
}
`;
